ilastik/utility/exportFile.py

```diff
@@ -447,17 +447,25 @@ def write_all(self, mode, compression=None):
         count = 0
         self.ExportProgress(0)
         if mode in ("h5", "hd5", "hdf5"):
-            with h5py.File(self.file_name, "w") as fout:
-                for table_name, table in self.table_dict.items():
-                    self._make_h5_dataset(
-                        fout,
-                        table_name,
-                        table,
-                        self.meta_dict.get(table_name, {}),
-                        compression if compression is not None else {},
-                    )
-                    count += 1
-                    self.ExportProgress(count * 100 / len(self.table_dict))
+            for libver in ["earliest", "v108"]:
+                try:
+                    with h5py.File(self.file_name, "w", libver=libver) as fout:
+                        for table_name, table in self.table_dict.items():
+                            self._make_h5_dataset(
+                                fout,
+                                table_name,
+                                table,
+                                self.meta_dict.get(table_name, {}),
+                                compression if compression is not None else {},
+                            )
+                            count += 1
+                            self.ExportProgress(count * 100 / len(self.table_dict))
+                except ValueError:
+                    logger.warning("Writing most compatible H5 failed, attempting HDF5 v1.8 format")
+                    count = 0
+                    self.ExportProgress(0)
+                    continue
```

Comment:
It would be nice to check here whether options of writing were exhausted - otherwise this routine will fail "quietly" with a warning and _not_ produce an output.